test/vecelement.jl

```diff
@@ -0,0 +1,19 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+typealias Vec{N,T} NTuple{N,Base.VecElement{T}}
+
+# Crash report for #15244 motivated this test.
+@generated function thrice_iota{N,T}(::Type{Vec{N,T}})
+    Expr(:tuple, [:(Base.VecElement(3*$i)) for i in 1:N]...)
```

Comment:
I'm having trouble getting this to work.  I'm getting complaints about the outer `$` not being inside quoted code.   (I have zero experience with generated functions.)